app/src/main/java/com/example/ivanvelazquez/proyectointent/InfoActivity.java

```diff
@@ -16,97 +18,95 @@
 import java.util.Locale;
 import java.util.Random;
 
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
 import static com.example.ivanvelazquez.proyectointent.ZooAnimales.EXTRA_ANIMAL;
 
 public class InfoActivity extends AppCompatActivity implements FavoritoView.Callback {
 
-    public static final String LINK= "link de mayor informacion";
+    public static final String LINK = "link de mayor informacion";
     private Animal animal;
-    private TextView especie;
-    private TextView info;
-    private ImageView foto;
-    private TextView horariosTv;
-    private TextView nombre;
-    private Button regresar;
-    private Button masInfo;
+    @BindView(R.id.tvEspecie)
+    TextView especie;
+    @BindView(R.id.tvInfo)
+    TextView info;
+    @BindView(R.id.ivFoto)
+    ImageView foto;
+    @BindView(R.id.TvHorarios)
+    TextView horariosTv;
+    @BindView(R.id.tvNombre)
+    TextView nombre;
+    @BindView(R.id.btnRegresar)
+    Button regresar;
+    @BindView(R.id.btnMasInfo)
+    Button masInfo;
+    @BindView(R.id.idFavourite)
+    FavoritoView favoritoView;
     private String idioma = Locale.getDefault().toString();
     private String url;
-    private FavoritoView favoritoView;
-    public static final String BACKGROUND="BACKGROUNDCOLOR";
-    public static final String BASEHOUR="BASEHOUR";
-    public int baseHour=0;
-    private int backgroundColor=0;
+    public int baseHour = 0;
+    private int backgroundColor = 0;
+    public static final String BACKGROUND = "BACKGROUNDCOLOR";
+    public static final String BASEHOUR = "BASEHOUR";
 
 
     @Override
-    protected void onCreate(Bundle savedInstanceState){
+    protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         restoreSavedInstance(savedInstanceState);
         setContentView(R.layout.activity_info_actiity);
         Bundle bundle = getIntent().getExtras();
-        horariosTv = findViewById(R.id.TvHorarios);
-        masInfo = findViewById(R.id.btnMasInfo);
-        especie = findViewById(R.id.tvEspecie);
-        info = findViewById(R.id.tvInfo);
+        ButterKnife.bind(this);
```

Comment:
Podes tener una clase comun abstracta que herede de `AppCompatActivity ` y ahi hacer el `ButterKnife.bind(this);` en el metodo `onCreate` . De esa manera, podes desde cualquier activity que implementes, podes heredar de esa clase comun y no te tenes que preocupar de hacer el bind con ButterKnife.

Reply:
Listoooo